test/datadog_logger_test.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:datadog_flutter/datadog_logger.dart';
+import 'package:datadog_flutter/src/channel.dart';
+import 'package:flutter/services.dart';
+import 'package:logging/logging.dart';
+import 'package:test/test.dart';
+
+import 'method_channel_helpers.dart';
+
+void main() {
+  group('DatadogLogger', () {
+    late DatadogLogger subject;
+    late List<MethodCall> calls;
+
+    setUp(() {
+      calls = [];
+      overrideMethodChannel(
```

Comment:
Instead of the mappings and method channel stubbing, what about making `_levelAsStatus` `@protected` and `@visibleForTesting`?

Reply:
I think I'd rather go the @protected route. The code is more legible and maintainable. And while this approach is more of an end to end test, it also involves stubbing and mocking, which could lead to false positives. This is a better candidate for unit testing than integration testing.

Reply:
Switched it up. Note that this sorta prevents us from reliably testing the `Level.OFF` case. The approach I took with that is IMO a bit less deterministic and more likely to cause false positives in the future, so if you want to ignore that case entirely let me know.

Reply:
@btrautmann is this method still necessary?

Reply:
Shoot, thought I responded but never actually submitted it:

Yeah, so I tried to call that out in the comments:
```
// Ignore calls. This prevents `MissingPluginException`s.
```
Basically without this, the tests still pass but for every test (i.e every `subject` instantiation) a `MissingPluginException` is thrown because there's no native side listening to the `MethodCall`s.